js/web/components/Hub.js

```diff
@@ -0,0 +1,268 @@
+import { useEffect, useMemo, useState, useContext } from 'react'
+import dynamic from 'next/dynamic'
+import Head from 'next/head'
+import { Box, } from '@mui/material'
+import { styled } from '@mui/system'
+import Hub from '@nina-protocol/nina-internal-sdk/esm/Hub'
+import Release from '@nina-protocol/nina-internal-sdk/esm/Release'
+import Audio from '@nina-protocol/nina-internal-sdk/esm/Audio'
+import { useSnackbar } from 'notistack'
+
+const Dots = dynamic(() => import('./Dots'))
+const HubHeader = dynamic(() => import('./HubHeader'))
+const HubCollaborators = dynamic(() => import('./HubCollaborators'))
+const HubToggle = dynamic(() => import('./HubToggle'))
+const HubReleases = dynamic(() => import('./HubReleases'))
+
+const HubComponent = ({ hubPubkey }) => {
+  const {
+    getHub,
+    hubState,
+    filterHubContentForHub,
+    filterHubCollaboratorsForHub,
+    hubContentState,
+  } = useContext(Hub.Context)
+
+  const { releaseState } = useContext(Release.Context)
+
+  const {resetQueueWithPlaylist} = useContext(Audio.Context)
+  const { enqueueSnackbar } = useSnackbar()
+
+  const [hubReleases, setHubReleases] = useState([])
+  const [releaseData, setReleaseData] = useState([])
+  const [collaboratorsData, setCollaboratorsData] = useState([])
+  const [view, setView] = useState('releases')
+  const [fetchedHubInfo, setFetchedHubInfo] = useState(false)
+  const [fetchedReleases, setFetchedReleases] = useState(false)
+  const [fetchedCollaborators, setFetchedCollaborators] = useState(false)
+  const hubData = useMemo(() => hubState[hubPubkey], [hubState, hubPubkey])
+
+  useEffect(() => {
+    if (!hubPubkey) {
+      setFetchedHubInfo(false)
+    }
+    getHub(hubPubkey)
+    if (hubPubkey) {
+      setFetchedHubInfo(true)
+    }
+  }, [hubPubkey])
+
+  useEffect(() => {
+    const [releases] = filterHubContentForHub(hubPubkey)
+    const collaborators = filterHubCollaboratorsForHub(hubPubkey)
+    setHubReleases(releases)
+    setCollaboratorsData(collaborators)
+    if (releases.length > 0) {
+      setFetchedReleases(true)
+    }
+    if (releases.length === 0) {
+      setFetchedReleases(false)
+    }
+    if (collaborators) {
+      setFetchedCollaborators(true)
+    }
+  }, [hubContentState])
+
+  useEffect(() => {
+    const data = hubReleases.map((hubRelease) => {
+      const releaseMetadata = releaseState.metadata[hubRelease.release]
+      releaseMetadata.releasePubkey = hubRelease.release
+      return releaseMetadata
+    })
+    setReleaseData(data)
+  }, [releaseState, hubReleases])
+
+  const releaseClickHandler = () => {
```

Comment:
i think these click handles can be turned into one function and driven by indexes